src/GovUk.Education.ExploreEducationStatistics.Data.Model/Migrations/Current_Routine_SelectObservationsByLocationCodes.sql

```diff
@@ -177,60 +197,21 @@ DECLARE
                 END
             SET @sql = left(@sql, len(@sql) - 3) + N') '
         END
-    SET @sql += N'GROUP BY o.Id ' +
-                'HAVING COUNT(DISTINCT ofi.FilterItemId) = (' +
-                '    SELECT COUNT(DISTINCT f.id) AS filterCount' +
-                '    FROM' +
-                '    FilterItem fi' +
-                '    JOIN FilterGroup fg ON fi.FilterGroupId = fg.Id' +
-                '    JOIN Filter f ON fg.FilterId = f.Id' +
-                '    WHERE fi.Id IN (SELECT id FROM @filterItemIds)' +
-                ') ' +
-                'ORDER BY o.Id;'
+
+    IF (@filterItemsExist = 1)
+        SET @sql += N'GROUP BY o.Id ' +
+                     'HAVING COUNT(DISTINCT ofi.FilterId) = @uniqueFiltersCount '
+
+    SET @sql += N'ORDER BY o.Id;'
+
     SET @paramDefinition = N'@subjectId uniqueidentifier,
-                           @filterItemIds IdListGuidType READONLY,
```

Comment:
I think the wrong version of the code is in this 'Current' version file?

Reply:
I think they're correct - they're the same as the latest updating migrations

Reply:
Got it, saw these params being removed and thought it must be the other version! I see now that we don't need them in the param definition for the `sp_executesql`. 🙂